### ISSUE TITLE
Site crashes when multiples courses get deleted

### ISSUE THREAD

**Issue by JurgenG** [chamilo/chamilo-lms#1482]
### Expected behavior / Resultado esperado / Résultat attendu
Given courses get deleted

### Actual behavior / Resultado real / Résultat réel
Error message:

Fatal error: Uncaught exception 'PDOException' with message 'SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`chamilo111`.`ticket_ticket`, CONSTRAINT `FK_EDE2C768591CC992` FOREIGN KEY (`course_id`) REFERENCES `course` (`id`))' in /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php on line 60

PDOException: SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`chamilo111`.`ticket_ticket`, CONSTRAINT `FK_EDE2C768591CC992` FOREIGN KEY (`course_id`) REFERENCES `course` (`id`)) in /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php on line 104

Call Stack:
    0.0014     439216   1. {main}() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:0
    0.1389    9021648   2. CourseManager::delete_course() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:285
    0.4240    9457360   3. Database::query() /var/www/1.11.chamilo.org/www/main/inc/lib/course.lib.php:2337
    0.4240    9457360   4. Doctrine\DBAL\Connection->executeQuery() /var/www/1.11.chamilo.org/www/main/inc/lib/database.lib.php:350
    0.4240    9457728   5. Doctrine\DBAL\Driver\PDOConnection->query() /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:833
    0.4240    9458248   6. PDO->query() /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:104

Doctrine\DBAL\Driver\PDOException: SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`chamilo111`.`ticket_ticket`, CONSTRAINT `FK_EDE2C768591CC992` FOREIGN KEY (`course_id`) REFERENCES `course` (`id`)) in /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php on line 106

Call Stack:
    0.0014     439216   1. {main}() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:0
    0.1389    9021648   2. CourseManager::delete_course() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:285
    0.4240    9457360   3. Database::query() /var/www/1.11.chamilo.org/www/main/inc/lib/course.lib.php:2337
    0.4240    9457360   4. Doctrine\DBAL\Connection->executeQuery() /var/www/1.11.chamilo.org/www/main/inc/lib/database.lib.php:350
    0.4240    9457728   5. Doctrine\DBAL\Driver\PDOConnection->query() /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:833

Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException: An exception occurred while executing 'DELETE FROM course WHERE code = 'CONTROLDIGITAL'':

SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`chamilo111`.`ticket_ticket`, CONSTRAINT `FK_EDE2C768591CC992` FOREIGN KEY (`course_id`) REFERENCES `course` (`id`)) in /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php on line 60

Call Stack:
    0.0014     439216   1. {main}() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:0
    0.1389    9021648   2. CourseManager::delete_course() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:285
    0.4240    9457360   3. Database::query() /var/www/1.11.chamilo.org/www/main/inc/lib/course.lib.php:2337
    0.4240    9457360   4. Doctrine\DBAL\Connection->executeQuery() /var/www/1.11.chamilo.org/www/main/inc/lib/database.lib.php:350

### Steps to reproduce / Pasos para reproducir / Étapes pour reproduire
1. Go to https://1.11.chamilo.org/main/admin/course_list.php?keyword=&submit=&_qf__search_simple=#
2. Select 2 or more courses
3. Go to the bottom and in the pulldown menu select "delete selected courses".
4. Major site crash.

Notice: when deleting 1 course, there is no error.
Notice 2: this works fine in 1.10

**Comment by ywarnier** [comment]
This is evidently linked to the recent addition of the ticket plugin into core, as entities, and the addition of foreign keys between ticket and course (and probably session). Julio, this should not be a restrictive foreign key... if the course or session are deleted, the ticket should replace the c_id or session_id by null and remain there (for logging purposes).

**Comment by JurgenG** [comment]
Hi,

This doesn't seem to be fixed on the 1.11.chamilo.org site...


```
Fatal error: Uncaught exception 'PDOException' with message 'SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`chamilo111`.`skill_rel_user`, CONSTRAINT `FK_79D3D95A591CC992` FOREIGN KEY (`course_id`) REFERENCES `course` (`id`))' in /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php on line 60

PDOException: SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`chamilo111`.`skill_rel_user`, CONSTRAINT `FK_79D3D95A591CC992` FOREIGN KEY (`course_id`) REFERENCES `course` (`id`)) in /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php on line 104

Call Stack:
    0.0021     440224   1. {main}() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:0
    0.3882   10140712   2. CourseManager::delete_course() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:285
    0.6730   10570368   3. Database::query() /var/www/1.11.chamilo.org/www/main/inc/lib/course.lib.php:2340
    0.6731   10570368   4. Doctrine\DBAL\Connection->executeQuery() /var/www/1.11.chamilo.org/www/main/inc/lib/database.lib.php:350
    0.6731   10570736   5. Doctrine\DBAL\Driver\PDOConnection->query() /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:833
    0.6731   10571256   6. PDO->query() /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:104

Doctrine\DBAL\Driver\PDOException: SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`chamilo111`.`skill_rel_user`, CONSTRAINT `FK_79D3D95A591CC992` FOREIGN KEY (`course_id`) REFERENCES `course` (`id`)) in /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php on line 106

Call Stack:
    0.0021     440224   1. {main}() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:0
    0.3882   10140712   2. CourseManager::delete_course() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:285
    0.6730   10570368   3. Database::query() /var/www/1.11.chamilo.org/www/main/inc/lib/course.lib.php:2340
    0.6731   10570368   4. Doctrine\DBAL\Connection->executeQuery() /var/www/1.11.chamilo.org/www/main/inc/lib/database.lib.php:350
    0.6731   10570736   5. Doctrine\DBAL\Driver\PDOConnection->query() /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:833

Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException: An exception occurred while executing 'DELETE FROM course WHERE code = 'BRASILTESTDESKILL'':

SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`chamilo111`.`skill_rel_user`, CONSTRAINT `FK_79D3D95A591CC992` FOREIGN KEY (`course_id`) REFERENCES `course` (`id`)) in /var/www/1.11.chamilo.org/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php on line 60

Call Stack:
    0.0021     440224   1. {main}() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:0
    0.3882   10140712   2. CourseManager::delete_course() /var/www/1.11.chamilo.org/www/main/admin/course_list.php:285
    0.6730   10570368   3. Database::query() /var/www/1.11.chamilo.org/www/main/inc/lib/course.lib.php:2340
    0.6731   10570368   4. Doctrine\DBAL\Connection->executeQuery() /var/www/1.11.chamilo.org/www/main/inc/lib/database.lib.php:350


```

**Comment by ywarnier** [comment]
That's because it's the kind of update that doesn't work directly through a git update. I'll reinstall the 1.11 portal shortly.

**Comment by ywarnier** [comment]
Seems to be fixed in https://1.11.chamilo.org/ now (I also created a ticket linked to one of the courses before deleting it).

**Comment by JurgenG** [comment]
Hi,

it's working now.
